sql/driver/driver_it_test.go

```diff
@@ -0,0 +1,304 @@
+		for rows.Next() {
+			if err := rows.Scan(&k, &vVarchar, &vTinyInt, &vSmallInt, &vInteger, &vBigInt, &vBool, &vReal, &vDouble); err != nil {
+				t.Fatal(err)
+			}
+			vs = append(vs, vVarchar, vTinyInt, vSmallInt, vInteger, vBigInt, vBool, vReal, vDouble)
+		}
+		target := []interface{}{"hello", int8(-128), int16(32767), int32(-27), int64(38), true, float32(-5.32), 12.789}
+		assert.Equal(t, target, vs)
+	})
+}
+
+func TestSQLQueryWithCursorBufferSize(t *testing.T) {
+	fn := func(i int) interface{} { return int32(i) }
+	ctx := driver.WithCursorBufferSize(context.Background(), 10)
+	testSQLQuery(t, ctx, "int", "int", fn, fn)
+}
+
+func TestSQLQueryWithQueryTimeout(t *testing.T) {
+	fn := func(i int) interface{} { return int32(i) }
+	ctx := driver.WithQueryTimeout(context.Background(), 5*time.Second)
+	testSQLQuery(t, ctx, "int", "int", fn, fn)
+}
+
+func testSQLQuery(t *testing.T, ctx context.Context, keyFmt, valueFmt string, keyFn, valueFn func(i int) interface{}) {
+	it.SQLTester(t, func(t *testing.T, client *hz.Client, config *hz.Config, m *hz.Map, mapName string) {
+		const rowCount = 50
+		target, err := populateMap(m, rowCount, keyFn, valueFn)
+		if err != nil {
+			t.Fatal(err)
+		}
+		dsn := makeDSN(config)
+		db, err := sql.Open("hazelcast", dsn)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ms := createMappingStr(mapName, keyFmt, valueFmt)
+		if err := createMapping(t, db, ms); err != nil {
+			t.Fatal(err)
+		}
+		defer db.Close()
+		query := fmt.Sprintf(`SELECT __key, this FROM "%s" ORDER BY __key`, mapName)
+		rows, err := db.QueryContext(ctx, query)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer rows.Close()
+		entries := make([]types.Entry, len(target))
+		var i int
+		for rows.Next() {
+			if err := rows.Scan(&entries[i].Key, &entries[i].Value); err != nil {
+				log.Fatal(err)
+			}
+			i++
+		}
+		assert.Equal(t, target, entries)
+	})
+}
+
+func createMapping(t *testing.T, db *sql.DB, mapping string) error {
+	t.Logf("mapping: %s", mapping)
+	_, err := db.Exec(mapping)
+	return err
+}
+
+func createMappingStr(mapName, keyFmt, valueFmt string) string {
+	return fmt.Sprintf(`
+        CREATE MAPPING "%s"
+        TYPE IMAP 
+        OPTIONS (
+            'keyFormat' = '%s',
+            'valueFormat' = '%s'
+        )
+`, mapName, keyFmt, valueFmt)
+}
+
+func populateMap(m *hz.Map, count int, keyFn, valueFn func(i int) interface{}) ([]types.Entry, error) {
+	entries := make([]types.Entry, count)
+	for i := 0; i < count; i++ {
+		entries[i] = types.Entry{
+			Key:   keyFn(i),
+			Value: valueFn(i),
+		}
+	}
+	if err := m.PutAll(context.Background(), entries...); err != nil {
+		return nil, err
+	}
+	return entries, nil
+}
+
+func makeDSN(config *hz.Config) string {
+	ll := logger.InfoLevel
+	if it.TraceLoggingEnabled() {
+		ll = logger.TraceLevel
+	}
+	return fmt.Sprintf("%s;cluster.name=%s;cluster.unisocket=%t;logger.level=%s", config.Cluster.Network.Addresses[0], config.Cluster.Name, config.Cluster.Unisocket, ll)
+}
```

Comment:
I would like to see that we can decode every column type, to see that we are compatible with the server. Similar to https://github.com/hazelcast/hazelcast-python-client/blob/master/tests/integration/backward_compatible/sql_test.py#L631

Reply:
Adde more tests at: https://github.com/hazelcast/hazelcast-go-client/pull/695/commits/db347f022df35b809058a714e6bb42db3964b9d2 All column types should be tested now.